src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -722,6 +722,10 @@ class MergeTreeData : public IStorage, public WithMutableContext
     /// Makes sense only for ordinary MergeTree engines because for them block numbering doesn't depend on partition.
     std::optional<Int64> getMinPartDataVersion() const;
 
+    /// Is the disk included in the scope for searching [detached] parts
+    ///   sometimes it is better to bypass a disk e.g. to avoid interactions with a remote storage
+    bool lookOnDisk(DiskPtr disk) const;
```

Comment:
I would make the method name a bit more descriptive (e.g, `shouldSearchForPartsOnDisk`)